angular/lib/src/runtime/errors.dart

```diff
@@ -0,0 +1,18 @@
+/// **INTERNAL ONLY**.
+library angular.src.runtime.errors;
+
+/// Thrown by the framework when an event handler expects an incorrect type.
+class InvalidHandlerError extends TypeError {
+  final Type expected;
+  final Type actual;
+  final Object context;
+
+  InvalidHandlerError(this.expected, this.actual, this.context);
+
+  @override
+  String toString() =>
+      'Invalid event handler on $context: Expected an event of type $expected '
```

Comment:
Saying the event handler is invalid isn't necessarily correct. Many of the errors I fixed were caused by emitting the wrong types from the stream the handler was registered to.

For example, a handler of type `(List<Foo>) => void` was registered to a stream of type `Stream<List<Foo>>`, but the client code was actually adding values of type `List` to the stream.

I've forgotten, will this sort of invalid implicit cast become an error as well? If so then I guess this message is correct, as it's the only way we'll get here.

Reply:
> Saying the event handler is invalid isn't necessarily correct. Many of the errors I fixed were caused by emitting the wrong types from the stream the handler was registered to.

Reply:
Yeah thinking about this more it seems like this _could_ provide a misleading error _until_ invalid implicit casts become a run-time error, but after that it's correct.

Are you fine writing it for the future now? Or do you want an intermediate message?